### DISCUSSION TITLE
a problem with proxy of 'CERT_HAS_EXPIRED'

### DISCUSSION THREAD

**Discussion by lemondreamtobe** [chimurai/http-proxy-middleware#559] [Q&A]
Today, I get a problem with http-proxy-middleware.
show my code follow:
```
const proxy = require('http-proxy-middleware');

module.exports = function (app) {
  app.use(
    proxy('/api/weather', {
      target: 'https://restapi.amap.com/v3/weather/weatherInfo',
      changeOrigin: true,
      secure: false,
      pathRewrite: {
        '^/api/weather': ''
      }
    })
  );
...
```
and the result of web browser
![image](https://user-images.githubusercontent.com/22766436/55303503-3ea09c80-5479-11e9-9203-fc764304ef8c.png)
![image](https://user-images.githubusercontent.com/22766436/55303511-43fde700-5479-11e9-9c19-7a3a36f20ec4.png)

and the log of server
![image](https://user-images.githubusercontent.com/22766436/55303533-5d9f2e80-5479-11e9-8834-bb4a7412ecfb.png)

so how can i fix it?

**Comment by chimurai** [comment]
Can you try adding option: `agent: https.globalAgent`?

See [http-to-https](https://github.com/nodejitsu/node-http-proxy/blob/f345a1ac2dde1884e72b952a685a0a1796059f14/examples/http/proxy-http-to-https.js#L40) example

https://github.com/nodejitsu/node-http-proxy/blob/master/examples/http/proxy-http-to-https.js#L40

**Comment by lemondreamtobe** [comment]
![image](https://user-images.githubusercontent.com/22766436/55526897-6423e500-56c9-11e9-83fb-2ee04e39d9fd.png)
Do you mean that?

**Comment by lemondreamtobe** [comment]
`
 app.use(
    proxy('/api/weather', {
      target: 'https://restapi.amap.com/v3/weather/weatherInfo',
      changeOrigin: true,
      secure: false,
      agent: https.globalAgent,
      pathRewrite: {
        '^/api/weather': ''
      }
    })
  );
`
